packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -427,16 +433,40 @@ export class PartialSequenceLengths {
     private static insertSegment(
         combinedPartialLengths: PartialSequenceLengths,
         segment: ISegment,
-        removalInfo?: IRemovalInfo) {
-        const isLocal = (removalInfo === undefined && segment.seq === UnassignedSequenceNumber)
-            || (removalInfo !== undefined && segment.removedSeq === UnassignedSequenceNumber);
+        removalInfo?: IRemovalInfo,
+        moveInfo?: IMoveInfo) {
+        const isLocal =
+            (removalInfo === undefined && moveInfo === undefined && segment.seq === UnassignedSequenceNumber)
+            || (removalInfo !== undefined && segment.removedSeq === UnassignedSequenceNumber)
+            || (moveInfo !== undefined && segment.movedSeq === UnassignedSequenceNumber);
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
         let seqOrLocalSeq = isLocal ? segment.localSeq! : segment.seq!;
         let segmentLen = segment.cachedLength;
         let clientId = segment.clientId;
         let removeClientOverlap: number[] | undefined;
 
-        if (removalInfo) {
+        if (removalInfo && moveInfo) {
+            const removalIsLocal = removalInfo.removedSeq === UnassignedSequenceNumber;
+            const moveIsLocal = moveInfo.movedSeq === UnassignedSequenceNumber;
+
+            assert(!moveIsLocal && !removalIsLocal || moveIsLocal !== removalIsLocal, "");
+
+            if (removalIsLocal) {
+                seqOrLocalSeq = moveInfo.movedSeq;
```

Comment:
seems weird that this is conditioned on removal but then grabs seq from moveInfo.

I see there's some todos later--rather than overcomment I'll assume there's some stuff that needs to be worked out here. It would be great to not get combinatorial explosion but I'm not sure I see a great way to do that. Might also be interesting to check out coverage stats on these lines with the partial lengths unit test suite.